app/controllers/buses_controller.rb

```diff
@@ -0,0 +1,53 @@
+require 'net/http'
+require 'json'
+
+class BusesController < ApplicationController
+  def index
+  end
+
+  def stop
+    stop = params[:stop]
+    @arrivals = fetch_arrivals([stop])
+    render :arrivals
+  end
+
+  def postcode
+    postcode = params[:postcode]
+    lat, lon = fetch_postcode_coordinates(postcode)
+    stops = fetch_nearest_stops(lat, lon)
+    @arrivals = fetch_arrivals(stops)
+    render :arrivals
+  end
+
+  private
+
+  def fetch_arrivals(stops)
+    arrivals = []
+
+    stops.each do |stop|
+      uri = URI("https://api.tfl.gov.uk/StopPoint/#{stop}/Arrivals")
+      res = Net::HTTP.get(uri)
+      buses = JSON.parse(res)
+      buses
+        .sort_by! { |x| x['timeToStation'].to_i }
+        .slice(0, 5)
```

Comment:
You could also use .first(5), not much diference

Reply:
There are two options: use `first` with reassignment (since it doesn't mutate)
```rb
buses = buses
  .sort_by! { }
  .first(5)
```
or use `slice` but in a way that actually works (the current code doesn't do anything because `slice` also doesn't mutate):
```rb
buses
  .sort_by! { }
  .slice!(5..)
```

Reply:
or option 3 is to combine with the previous line which is probably my favourite
```rb
buses = JSON.parse(res)
  .sort_by { |x| x['timeToStation'].to_i }
  .first(5)
```